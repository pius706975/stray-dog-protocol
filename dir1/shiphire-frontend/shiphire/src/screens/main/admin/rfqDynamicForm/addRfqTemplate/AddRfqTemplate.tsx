import React from 'react';
import { Text, View } from 'react-native-ui-lib';
import {
    AddRfqTemplateProps,
    AddRFQTemplateRequest,
} from '../../../../../types';
import {
    CustomText,
    ScreenLayout,
    TextInput,
    TextInputError,
} from '../../../../../components';
import { Formik, FormikHelpers, FormikProps } from 'formik';
import * as yup from 'yup';
import { Color, FontFamily, FontSize } from '../../../../../configs';
import CustomButton from '../../../../../components/Button';
import {
    useActiveDynamicInput,
    useAddRfqTemplate,
    useGetShipCategories,
} from '../../../../../hooks';
import { useDispatch } from 'react-redux';
import { modalSlice } from '../../../../../slices';
import DropDownPicker from 'react-native-dropdown-picker';
import { useIsFocused } from '@react-navigation/native';
import { handleAxiosError } from '../../../../../utils';

const AddRfqTemplate: React.FC<AddRfqTemplateProps> = ({ navigation }) => {
    const dispatch = useDispatch();
    const { hideModal, showModal } = modalSlice.actions;
    const [fieldType, setFieldType] = React.useState('');
    const [open, setOpen] = React.useState(false);
    const [items, setItems] = React.useState([
        {
            label: 'Default',
            value: 'default',
            testID: 'test-default',
        },
    ]);
    const mutationAddRfqTemplate = useAddRfqTemplate();
    const mutationgetShipCategories = useGetShipCategories();
    const isFocused = useIsFocused();

    React.useEffect(() => {
        if (isFocused) {
            mutationgetShipCategories.mutate(undefined, {
                onSuccess: resp => {
                    const shipCategories = resp.data.data;
                    shipCategories.map(item => {
                        setItems(prev => [
                            ...prev,
                            {
                                label: item.name,
                                value: item.name,
                                testID: `item-${item.name}`,
                            },
                        ]);
                    });
                },
                onError: err => {
                    handleAxiosError(err);
                },
            });
        }
    }, [isFocused]);

    const addRfqTemplateValues: AddRFQTemplateRequest = {
        shipType: '',
    };

    const addRfqTemplateSchema = yup.object().shape({
        shipType: yup.string().required('Ship Type is required'),
    });

    const handleAddRfqTemplate = (
        values: AddRFQTemplateRequest,
        actions: FormikHelpers<AddRFQTemplateRequest>,
    ) => {
        const payload = {
            templateType: `${values.shipType}Rfq`,
        };
        mutationAddRfqTemplate.mutate(payload, {
            onSuccess: () => {
                dispatch(
                    showModal({
                        status: 'success',
                        text: 'Add RFQ Template Success',
                    }),
                );

                setTimeout(() => {
                    dispatch(hideModal());
                }, 3000);
                actions.setSubmitting(false);
                navigation.navigate('RFQTemplateManagement');
            },
            onError: err => {
                dispatch(
                    showModal({
                        status: 'failed',
                        text: 'Add RFQ Template Failed/Duplicate Template',
                    }),
                );

                setTimeout(() => {
                    dispatch(hideModal());
                }, 3000);
                actions.setSubmitting(false);
            },
        });
    };
    return (
        <ScreenLayout
            backgroundColor="light"
            testId="AddRfqTemplate"
            gap={10}
            padding={10}>
            <View style={{ marginVertical: 8, alignItems: 'center' }}>
                <CustomText
                    fontSize="sm"
                    fontFamily="semiBold"
                    color="darkTextColor">
                    Please enter the ship type for the new RFQ template
                </CustomText>
            </View>
            <Formik
                initialValues={addRfqTemplateValues}
                onSubmit={handleAddRfqTemplate}
                validationSchema={addRfqTemplateSchema}>
                {({
                    handleBlur,
                    handleChange,
                    handleSubmit,
                    values,
                    touched,
                    errors,
                    isSubmitting,
                    setFieldValue,
                }: FormikProps<AddRFQTemplateRequest>) => (
                    <>
                        <View style={{ paddingLeft: 10 }}>
                            <Text
                                style={{
                                    fontFamily: FontFamily.medium,
                                    fontSize: FontSize.xl,
                                    color: Color.primaryColor,
                                    marginBottom: 10,
                                }}>
                                Select Template Type:
                            </Text>
                            <DropDownPicker
                                testID="template-picker"
                                open={open}
                                value={fieldType}
                                items={items}
                                setOpen={setOpen}
                                onPress={setOpen}
                                setValue={setFieldType}
                                setItems={setItems}
                                mode="SIMPLE"
                                listMode="MODAL"
                                searchable={false}
                                modalTitle="Select an item"
                                modalProps={{
                                    animationType: 'slide',
                                }}
                                style={{
                                    backgroundColor: Color.bgColor,
                                    borderColor: Color.bgNeutralColor,
                                    elevation: 2,
                                }}
                                placeholderStyle={{
                                    color: Color.primaryColor,
                                    fontFamily: FontFamily.regular,
                                }}
                                modalContentContainerStyle={{
                                    backgroundColor: Color.bgColor,
                                }}
                                modalTitleStyle={{
                                    fontFamily: FontFamily.regular,
                                }}
                                labelStyle={{
                                    color: Color.primaryColor,
                                    fontFamily: FontFamily.regular,
                                }}
                                listItemContainerStyle={{
                                    borderBottomWidth: 1,
                                    marginVertical: 5,
                                    borderColor: Color.bgNeutralColor,
                                    // elevation: 1,
                                }}
                                listItemLabelStyle={{
                                    color: Color.primaryColor,
                                    fontFamily: FontFamily.regular,
                                }}
                                selectedItemContainerStyle={{
                                    backgroundColor: Color.secColor,
                                }}
                                selectedItemLabelStyle={{
                                    fontFamily: FontFamily.bold,
                                    color: Color.bgColor,
                                }}
                                onSelectItem={e =>
                                    setFieldValue('shipType', e.value)
                                }
                            />
                        </View>
                        <View
                            flexG-1
                            marginB-30
                            style={{ justifyContent: 'flex-end' }}>
                            <CustomButton
                                testID="submit-btn"
                                title="Add New RFQ Template"
                                isSubmitting={isSubmitting}
                                onSubmit={() => handleSubmit()}
                            />
                        </View>
                    </>
                )}
            </Formik>
        </ScreenLayout>
    );
};

export default AddRfqTemplate;
