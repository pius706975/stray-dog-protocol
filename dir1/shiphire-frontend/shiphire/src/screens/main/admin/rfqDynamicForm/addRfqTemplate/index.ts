export { default as AddRfqTemplate } from './AddRfqTemplate';
