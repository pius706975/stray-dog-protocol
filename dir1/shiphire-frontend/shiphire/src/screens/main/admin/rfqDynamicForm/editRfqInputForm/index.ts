export { default as EditRFQInputForm } from './EditRFQInputForm';
