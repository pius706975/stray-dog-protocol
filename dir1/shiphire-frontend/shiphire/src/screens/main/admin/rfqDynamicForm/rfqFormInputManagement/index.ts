export { default as RFQFormInputManagement } from './RFQFormInputManagement';
