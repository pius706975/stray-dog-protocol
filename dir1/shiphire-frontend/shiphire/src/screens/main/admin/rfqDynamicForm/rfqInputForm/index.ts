export { default as RFQInputForm } from './RFQInputForm';
