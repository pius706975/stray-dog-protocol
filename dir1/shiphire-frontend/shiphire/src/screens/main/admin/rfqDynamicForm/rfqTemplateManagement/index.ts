export { default as RFQTemplateManagement } from './RFQTemplateManagement';
