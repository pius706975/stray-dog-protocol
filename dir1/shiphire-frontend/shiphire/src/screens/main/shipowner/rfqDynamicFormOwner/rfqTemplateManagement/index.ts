export { default as RFQTemplateOwnerManagement } from './RFQTemplateOwnerManagement';
