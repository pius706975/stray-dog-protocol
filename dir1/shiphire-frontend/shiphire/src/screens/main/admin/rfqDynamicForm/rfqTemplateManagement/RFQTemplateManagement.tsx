import React from 'react';
import { Pressable } from 'react-native';
import { FlatList } from 'react-native-gesture-handler';
import { Badge, View } from 'react-native-ui-lib';
import { Button, CustomText } from '../../../../../components';
import { Color, PlusIcon, RightIcon } from '../../../../../configs';
import {
    DynamicFormRFQ,
    RFQTemplateManagementProps,
} from '../../../../../types';
import {
    useActiveDynamicForm,
    useGetAllTemplateRFQForm,
} from '../../../../../hooks';
import { useIsFocused } from '@react-navigation/native';
import { ConfirmModal } from '../rfqFormInputManagement/components';
import { useDispatch } from 'react-redux';
import { modalSlice } from '../../../../../slices';

const RFQTemplateManagement: React.FC<RFQTemplateManagementProps> = ({
    navigation,
}) => {
    const templates = [
        {
            id: '1',
            title: 'Default RFQ Form',
            templateType: 'generalRfqDefault',
            formType: 'rfqForm',
        },
        {
            id: '2',
            title: 'Default Barge RFQ Form',
            templateType: 'bargeRfqDefault',
            formType: 'rfqForm',
        },
        {
            id: '3',
            title: 'Default Ferry RFQ Form',
            templateType: 'ferryRfqDefault',
            formType: 'rfqForm',
        },
        {
            id: '4',
            title: 'Default Tugboat RFQ Form',
            templateType: 'tugboatRfqDefault',
            formType: 'rfqForm',
        },
    ];
    const isFocused = useIsFocused();
    const mutationGetAllTemplateRFQForm = useGetAllTemplateRFQForm();
    const mutationActiveDynamicFormRFQ = useActiveDynamicForm();
    const dispatch = useDispatch();
    const { hideModal, showModal } = modalSlice.actions;
    const [showConfirmationModal, setShowConfirmationModal] =
        React.useState(false);
    const [selectedDynamicInputId, setSelectedDynamicInputId] =
        React.useState(null);
    const [dynamicFormData, setDynamicFormData] =
        React.useState<DynamicFormRFQ[]>();

    React.useEffect(() => {
        mutationGetAllTemplateRFQForm.mutate(undefined, {
            onSuccess: resp => {
                setDynamicFormData(resp.data.data);
                console.log('resp.data.data', resp.data.data);
            },
            onError: err => {
                console.log(err);
            },
        });
    }, []);

    React.useEffect(() => {
        isFocused &&
            mutationGetAllTemplateRFQForm.mutate(undefined, {
                onSuccess: resp => {
                    setDynamicFormData(resp.data.data);
                },
            });
    }, [isFocused]);

    React.useLayoutEffect(() => {
        navigation.setOptions({
            headerRight: () => (
                <Pressable
                    testID="btn-add-template"
                    onPress={() => navigation.navigate('AddRfqTemplate')}>
                    <PlusIcon />
                </Pressable>
            ),
        });
    });

    const handleConfirmDeactivation = () => {
        if (!selectedDynamicInputId) {
            console.error('Selected dynamic input ID is not set.');
            return;
        }

        mutationActiveDynamicFormRFQ.mutate(
            {
                id: selectedDynamicInputId,
            },
            {
                onSuccess: resp => {
                    dispatch(
                        showModal({
                            status: 'success',
                            text: 'Update Template RFQ Form Status Success',
                        }),
                    );
                    setTimeout(() => {
                        dispatch(hideModal());
                    }, 4000);
                    mutationGetAllTemplateRFQForm.mutate(undefined, {
                        onSuccess: resp => {
                            setDynamicFormData(resp.data.data);
                        },
                    });
                },
                onError: err => {
                    dispatch(
                        showModal({
                            status: 'failed',
                            text: 'Update Template RFQ Form Status Failed',
                        }),
                    );
                    setTimeout(() => {
                        dispatch(hideModal());
                    }, 4000);
                },
            },
        );

        setShowConfirmationModal(false);
    };

    const handleCancelDeactivation = () => {
        // Close the confirmation modal
        setShowConfirmationModal(false);
    };

    const getConfirmationMessage = active => {
        return active
            ? 'Are you sure you want to deactivate?'
            : 'Are you sure you want to activate?';
    };

    const renderRFQTemplate = ({ item }) => {
        const { title, templateType, formType } = item;
        const deactivateDynamicInput = id => {
            setSelectedDynamicInputId(id);
            setShowConfirmationModal(true);
        };
        const activeDynamicInput = id => {
            setSelectedDynamicInputId(id);
            setShowConfirmationModal(true);
        };

        return (
            <View
                row
                style={{
                    justifyContent: 'space-between',
                    borderBottomWidth: 0.5,
                }}>
                <Pressable
                    style={{ flex: 1 }}
                    onPress={() => {
                        navigation.navigate('RFQFormInputManagement', {
                            templateType,
                            formType,
                        });
                    }}>
                    <View
                        style={{
                            padding: 10,
                            margin: 5,
                            // flexDirection: 'row',
                            justifyContent: 'space-between',
                            // alignItems: 'center',
                        }}>
                        <View
                            style={{
                                flexDirection: 'row',
                            }}>
                            <View
                                style={{
                                    justifyContent: 'flex-start',
                                    padding: 5,
                                    paddingRight: 10,
                                    paddingLeft: 0,
                                }}></View>

                            <CustomText
                                fontFamily="regular"
                                fontSize="md"
                                color="darkTextColor">
                                {templateType
                                    .split(/(?=[A-Z])/)
                                    .map(
                                        word =>
                                            word.charAt(0).toUpperCase() +
                                            word.slice(1),
                                    )
                                    .join(' ')}
                            </CustomText>
                        </View>
                        {/* <RightIcon /> */}
                        <View marginT-5>
                            {item.active ? (
                                <Badge
                                    marginL-10
                                    label="Active"
                                    borderRadius={4}
                                    backgroundColor={Color.primaryColor}
                                    size={16}
                                />
                            ) : (
                                <Badge
                                    marginL-10
                                    label="Inactive"
                                    borderRadius={4}
                                    backgroundColor={Color.errorColor}
                                    size={16}
                                />
                            )}
                        </View>
                    </View>
                </Pressable>
                <View
                    row
                    style={{
                        alignItems: 'center',
                    }}>
                    <View marginR-5>
                        {/* <Button
                            title="Edit"
                            color="success"
                            onSubmit={() =>
                                navigation.navigate('EditRFQInputForm', {
                                    _id: item._id,
                                    templateType,
                                    formType,
                                })
                            }
                        /> */}
                    </View>
                    <View marginR-5>
                        {item.active ? (
                            <Button
                                testID="btn-confirm-deactive"
                                title="Deactive"
                                color="error"
                                onSubmit={() =>
                                    deactivateDynamicInput(item._id)
                                }
                            />
                        ) : (
                            <Button
                                testID="btn-confirm-active"
                                title="Active"
                                color="primary"
                                onSubmit={() => activeDynamicInput(item._id)}
                            />
                        )}
                    </View>
                </View>
            </View>
        );
    };

    return (
        <View style={{ flex: 1 }}>
            {dynamicFormData && dynamicFormData.length === 0 ? (
                <View
                    style={{
                        flex: 1,
                        justifyContent: 'center',
                        alignItems: 'center',
                    }}>
                    <CustomText
                        color="darkTextColor"
                        fontSize="lg"
                        fontFamily="regular">
                        No RFQ Templates available.
                    </CustomText>
                </View>
            ) : (
                <FlatList
                    style={{ flex: 1 }}
                    data={['']}
                    keyExtractor={() => 'dummyKey'}
                    showsVerticalScrollIndicator={false}
                    renderItem={() => (
                        <View
                            style={{
                                margin: 6,
                                borderRadius: 8,
                            }}>
                            <FlatList
                                ItemSeparatorComponent={() => <View />}
                                data={dynamicFormData}
                                renderItem={renderRFQTemplate}
                                keyExtractor={item => item._id}
                                onEndReachedThreshold={0.1}
                            />
                        </View>
                    )}
                />
            )}
            {/* <View
                style={{
                    position: 'absolute',
                    bottom: 30,
                    right: 30,
                }}>
                <Button
                    title="New Template RFQ"
                    color="primary"
                    leftIcon={<PlusIcon />}
                    onSubmit={() => {
                        navigation.navigate('AddRfqTemplate');
                    }}
                />
            </View> */}
            <ConfirmModal
                isVisible={showConfirmationModal}
                onConfirm={handleConfirmDeactivation}
                onCancel={handleCancelDeactivation}
                confirmationMessage={getConfirmationMessage(
                    dynamicFormData &&
                        selectedDynamicInputId &&
                        dynamicFormData.find(
                            item => item._id === selectedDynamicInputId,
                        )?.active,
                )}
            />
        </View>
    );
};

export default RFQTemplateManagement;
