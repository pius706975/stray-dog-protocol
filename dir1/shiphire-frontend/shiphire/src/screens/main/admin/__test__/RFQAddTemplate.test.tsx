import { QueryClient, QueryClientProvider } from 'react-query';
import { AddRfqTemplate } from '../rfqDynamicForm/addRfqTemplate';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { MainAdminStackParamList } from '../../../../types';
import httpRequest from '../../../../services/api';
import MockAdapter from 'axios-mock-adapter';
import store from '../../../../store';
import { Provider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';
import { useAddRfqTemplate, useGetShipCategories } from '../../../../hooks';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { act } from 'react-test-renderer';
import { RFQTemplateManagement } from '../rfqDynamicForm/rfqTemplateManagement';

jest.mock('../../../../hooks');
const queryClient = new QueryClient();
const MainAdminStack = createNativeStackNavigator<MainAdminStackParamList>();

const mock = new MockAdapter(httpRequest);
const mockCategories = [
    {
        _id: '1',
        name: 'barge',
        __v: '1',
    },
    {
        _id: '2',
        name: 'tugboat',
        __v: '2',
    },
];
const adminStackMockComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainAdminStack.Navigator>
                    <MainAdminStack.Screen
                        name="AddRfqTemplate"
                        component={AddRfqTemplate}
                    />
                </MainAdminStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

const mockSubmitForm = () => {
    mock.onGet('/admin/save-rfq-template').reply(200, {
        message: 'success',
    });
    (useAddRfqTemplate as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onSuccess } = options;
            const mockResponseData = {
                message: 'success',
            };
            onSuccess(mockResponseData);
        }),
    }));
};
const mockSubmitFormError = () => {
    mock.onGet('/admin/save-rfq-template').reply(500, {
        message: 'failed',
    });
    (useAddRfqTemplate as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onError } = options;
            const mockResponseData = {
                message: 'failed',
            };
            onError(mockResponseData);
        }),
    }));
};

beforeEach(() => {
    mock.onGet('/admin/get-categories').reply(200, {
        message: 'success',
        data: mockCategories,
    });
    (useGetShipCategories as jest.Mock).mockImplementation(() => ({
        mutate: jest.fn().mockImplementation((data, options) => {
            const { onSuccess } = options;
            const mockResponseData = {
                data: {
                    data: mockCategories,
                },
            };
            onSuccess(mockResponseData);
        }),
    }));
});

describe('Testing RFQ add template screen', () => {
    describe('Snapshot testing', () => {
        it('should render RFQ add template screen correctly', () => {
            const rfqAddTemplate = render(adminStackMockComponent);
            expect(rfqAddTemplate).toMatchSnapshot();
        });
    });
    describe('unit testing', () => {
        it('should submit form add template test', async () => {
            mockSubmitForm();
            const { getByTestId } = render(adminStackMockComponent);
            const templatePicker = getByTestId('template-picker');
            act(() => {
                fireEvent.press(templatePicker);
            });
            const bargeItem = getByTestId('item-barge');
            fireEvent.press(bargeItem);

            const submitBtn = getByTestId('submit-btn');
            fireEvent.press(submitBtn);

            await waitFor(() => {
                const modalState = store.getState().modal;

                expect(modalState).toEqual({
                    visible: true,
                    status: 'success',
                    text: 'Add RFQ Template Success',
                });
            });
        });
        it('should show error when add template failed', async () => {
            mockSubmitFormError();
            const { getByTestId } = render(adminStackMockComponent);
            const templatePicker = getByTestId('template-picker');
            act(() => {
                fireEvent.press(templatePicker);
            });
            const bargeItem = getByTestId('item-barge');
            fireEvent.press(bargeItem);

            const submitBtn = getByTestId('submit-btn');
            fireEvent.press(submitBtn);

            await waitFor(() => {
                const modalState = store.getState().modal;

                expect(modalState).toEqual({
                    visible: true,
                    status: 'failed',
                    text: 'Add RFQ Template Failed/Duplicate Template',
                });
            });
        });
    });
});
