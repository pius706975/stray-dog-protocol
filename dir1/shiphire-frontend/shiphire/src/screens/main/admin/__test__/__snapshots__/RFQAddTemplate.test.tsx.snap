// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RFQ add template screen Snapshot testing should render RFQ add template screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="AddRfqTemplate"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": 10,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": 10,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "alignItems": "center",
                        "marginVertical": 8,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={0}
                    style={
                      {
                        "borderBottomWidth": 0,
                        "borderColor": undefined,
                        "color": "#4d5963",
                        "fontFamily": "Kanit-SemiBold",
                        "fontSize": 16,
                        "lineHeight": undefined,
                        "textAlign": undefined,
                      }
                    }
                  >
                    Please enter the ship type for the new RFQ template
                  </Text>
                </View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "paddingLeft": 10,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "color": "#20303C",
                          "writingDirection": "ltr",
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "color": "#216dab",
                          "fontFamily": "Kanit-Medium",
                          "fontSize": 22,
                          "marginBottom": 10,
                        },
                      ]
                    }
                  >
                    Select Template Type:
                  </Text>
                  <View
                    style={
                      [
                        {
                          "width": "100%",
                        },
                        {
                          "zIndex": 5000,
                        },
                        {},
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      forwardedRef={[Function]}
                      onClick={[Function]}
                      onLayout={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#f0f6fb",
                          "borderColor": "#dbdee0",
                          "borderRadius": 8,
                          "borderWidth": 1,
                          "elevation": 2,
                          "flexDirection": "row",
                          "justifyContent": "space-between",
                          "minHeight": 50,
                          "opacity": 1,
                          "paddingHorizontal": 10,
                          "paddingVertical": 3,
                          "width": "100%",
                          "zIndex": 5000,
                        }
                      }
                      testID="template-picker"
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#000",
                              "flex": 1,
                            },
                            {},
                            false,
                            {
                              "color": "#216dab",
                              "fontFamily": "Kanit-Regular",
                            },
                          ]
                        }
                      >
                        Select an item
                      </Text>
                      <View
                        style={
                          [
                            {
                              "marginLeft": 10,
                            },
                            {},
                          ]
                        }
                      >
                        <Image
                          source={
                            {
                              "testUri": "../../react-native-dropdown-picker/src/themes/light/icons/arrow-down.png",
                            }
                          }
                          style={
                            [
                              {
                                "height": 20,
                                "width": 20,
                              },
                              {},
                            ]
                          }
                        />
                      </View>
                    </View>
                    <Modal
                      animationType="slide"
                      hardwareAccelerated={false}
                      onRequestClose={[Function]}
                      presentationStyle="fullScreen"
                      visible={false}
                    />
                  </View>
                </View>
                <View
                  flexG-1={true}
                  marginB-30={true}
                  style={
                    [
                      undefined,
                      undefined,
                      {
                        "flexGrow": 1,
                      },
                      undefined,
                      {},
                      {
                        "marginBottom": 30,
                      },
                      {},
                      {
                        "justifyContent": "flex-end",
                      },
                    ]
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#216dab",
                        "borderRadius": 6,
                        "elevation": 2,
                        "flexDirection": "row",
                        "gap": 0,
                        "justifyContent": "center",
                        "paddingHorizontal": 10,
                        "paddingVertical": 6,
                      }
                    }
                    testID="submit-btn"
                  >
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#f0f6fb",
                          "fontFamily": "Kanit-Medium",
                          "fontSize": 18,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      Add New RFQ Template
                    </Text>
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "position": "absolute",
                            "right": 20,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="AddRfqTemplate"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
